code/postgres/src/main/scala/postgres/clients/PostgresClient.scala

```diff
@@ -43,9 +43,25 @@ trait UsingPostgres {
   sqlDateFormatter.setTimeZone(TimeZone.getTimeZone("UTC"))
 
   val log                                      = LoggerFactory.getLogger(getClass)
-  val mainSchema                               = "atomic"
+  val mainSchema                               = {
+                                                    val splitArray=parentTable.split("\\.")
+                                                    if (splitArray.length==2){
+                                                      parentTable.split("\\.")(0)
```

Comment:
you can simply add something like this  -

  val mainSchema = Try(parentTable.split("\\.")(0)).getOrElse("atomic")
  val coreTableName = Try(parentTable.split("\\.")(1)).getOrElse("events")

Reply:
I have also taken into consideration the length of the array after splitting the string